connectors/src/connectors/snowflake/temporal/activities.ts

```diff
@@ -66,6 +52,42 @@ export async function syncSnowflakeConnection(connectorId: ModelId) {
     }),
   ]);
 
+  const grantsRes = await fetchGrants({ credentials, connection });
+  if (grantsRes.isErr()) {
+    throw grantsRes.error;
+  }
+  const grantsCheck = areGrantsAreReadonly(grantsRes.value);
```

Comment:
Nit: why not make the call directly in `areGrantsAreReadonly` (also: `areGrantsAreReadonly` => `areGrantsReadonly`)

Reply:
Which call ? The fetch ?

Reply:
yep

Reply:
Yeah would make sense, we don't use the grants for anything else

Reply:
You can keep the method if you want, just make the call in the `areGrantsReadonly` method.

Reply:
I kept only one method (`isConnectionReadonly`)